jdaviz/configs/imviz/tests/test_catalogs.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+
+from astropy.io import fits
+
+'''
+This file tests four tasks of the catalogs.py plugin in Imviz.
+1. When no image/data is provided, the search does not crash
+- Additionally there are no results available as the search did not occur
+2. When an image/data is provided, specifically with no results being found,
+results are available and are set to 0
+3. When an image/data is provided, specifically with results being found,
+results are available and are set to > 0
+4. When the markers are cleared, there are no results available
+
+The data used for testing are provided in the comments below.
+
+Currently, these tests are driven by an SDSS search.
+As more catalogs are added to the plugin, each test will need to be
+specific based on the respective catalog.
+
+Additionally, tests that check for results > 0 are correct as the data was
+found from an SDSS server (and therefore should have results).
+However, the number of results is determined on the belief that the search
+is correct and the removal of the points outside the zoom limits is correctly done.
+This may need to instead be tested for in the future.
+'''
+
+
+# testing that the plugin search does not crash when no data/image is provided
+class TestCatalogs_NoImage():
+
+    def test_plugin(self, imviz_helper):
+        self.imviz = imviz_helper
+
+        catalogs_plugin = self.imviz.app.get_tray_item_from_name('imviz-catalogs')
+        catalogs_plugin.plugin_opened = True
+        # running the search without any data loaded into Imviz
+        catalogs_plugin.vue_do_search()
+
+        assert not catalogs_plugin.results_available
+
+
+# testing that every variable updates correctly when the image/data provided does not have results
+# data used: information based on the Imviz example notebook, variable "acs_47tuc_1"
+# https://github.com/spacetelescope/jdaviz/blob/main/notebooks/ImvizExample.ipynb
+class TestCatalogs_NoResults():
+
+    def test_plugin(self, imviz_helper):
+        arr = np.ones((2048, 4096))
+
+        # header is based on the data provided above
+        hdu1 = fits.ImageHDU(arr, name='SCI')
+        hdu1.header.update({'CTYPE1': 'RA---TAN-SIP',
+                            'CUNIT1': 'deg',
+                            'CD1_1': 1.0548878548304e-05,
+                            'CD1_2': 9.763832242379e-06,
+                            'CRPIX1': 2048.0,
+                            'CRVAL1': 6.0705364649855,
+                            'NAXIS1': 4096,
+                            'CTYPE2': 'DEC--TAN-SIP',
+                            'CUNIT2': 'deg',
+                            'CD2_1': 8.9680807000521e-06,
+                            'CD2_2': -9.9965364544771e-06,
+                            'CRPIX2': 1024.0,
+                            'CRVAL2': -71.880349631403,
+                            'NAXIS2': 2048})
+        imviz_helper.load_data(hdu1, data_label='no_results_data')
```

Comment:
Can we use one of the canned test cases that is much smaller in memory footprint? The ACS WCS here is incomplete anyway, since you are reconstructing SIP WCS without the SIP parts.

Assuming that this dummy test case actually does return a null result, so you'll have to play with it.

```suggestion
    def test_plugin_image_no_result(self, imviz_helper, image_2d_wcs):
        arr = np.ones((10, 10))
        ndd = NDData(arr, wcs=image_2d_wcs)

        imviz_helper.load_data(ndd, data_label='no_results_data')
```